src/database_observer.py

```diff
@@ -64,7 +64,7 @@ def _change_config(self, charm_state: CharmState) -> None:
             self._charm.unit.status = ops.MaintenanceStatus("Waiting for Synapse pebble")
             return
         try:
-            pebble.change_config(charm_state, container)
+            pebble.change_config(charm_state, container, is_leader=self._charm.unit.is_leader())
```

Comment:
IIUC, this line relates juju leader with synapse main process. However, juju leadership is only guaranteed to be valid for [30 seconds](https://discourse.charmhub.io/t/leadership-hook-tools/1124), so whenever that happens, I understand that the Synapse main process and worker process will be updated and restarted?

Reply:
Actually, good point. I need to add an observer for the leader-elected ;-)

Reply:
Could relating juju leaders and synapse main/workers create the possibility of downtime? (as there could be many restarts/changes)

Reply:
Unfortunately yes. Any changes in the horizontal scaling regarding who is the main and who is a worker require a complete restart. Same for scaling down/up.

Reply:
This is a very interesting case. Pinging @gregory-schiano and @mthaddon , as it is a situation we will have to handle for all services with this "main/worker" situation.

Reply:
Thanks for raising this, I'll change the approach now :)